avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializer.java

```diff
@@ -53,6 +54,11 @@ public Object deserialize(String s, byte[] bytes) {
     return deserialize(bytes);
   }
 
+  /**
+   * Pass a reader schema to get an Avro projection
+   */
+  public Object deserialize(String s, byte[] bytes, Schema readerSchema) { return deserialize(bytes, readerSchema); }
```

Comment:
Ok, so that makes a lot more sense since in your use case you are just making the call directly. I was (by default) thinking about this from the perspective of a user of Kafka consumers.

This use case makes sense, and I think it's great to support it directly in this code -- there's definitely no need to duplicate any of this. So I guess my impression now is that this patch will support your use case, we should leave in "dynamic" lookup of the reader schema to support using the same decoder/deserializer for both key and value simultaneously, and while this has no direct impact on regular Kafka users of the deserializer, it does set the stage for introducing an option to specify a different reader schema for those users. Is that right?